Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtension.cpp

```diff
@@ -110,11 +112,47 @@ using namespace WebKit;
  *   // ...
  * }
  * ```
+ *
+ * To send user data related to a particular #WebKitWebView, you have also the possibility of using
+ * the #WebKitWebView::initialize-web-extensions signal in combination with the
+ * webkit_web_view_set_web_extensions_initialization_user_data():
+ *
+ * ```c
+ * g_signal_connect(view, "initialize-web-extensions", G_CALLBACK(initialize_web_extensions), this);
+ *
+ * static void
+ * initialize_web_extensions (WebKitWebView* view, gpointer* user_data)
+ * {
+ *     webkit_web_view_set_web_extensions_initialization_user_data(view, g_variant_new_uint32 (unique_id++));
+ * }
+ * ```
+ *
+ * The user data can be recovered later by using the #WebKitWebExtension::initialize-web-extensions signal
+ * in the extension code:
+ *
+ * ```c
+ * static void
+ * web_extension_initialize_callback (WebKitWebExtension *extension,
+ *                                    GVariant           *init_data,
+ *                                    gpointer            user_data)
+ * {
+ *     g_print ("Page %d created for %s\n",
+ *              webkit_web_page_get_id (web_page),
+ *              webkit_web_page_get_uri (web_page));
```

Comment:
This won't build because you don't actually have any way to get `web_page` here. That's actually my main concern with this pull request....